sys/targets/targets.go

```diff
@@ -464,6 +489,7 @@ func initTarget(target *Target, OS, arch string) {
 		target.CPP = target.CCompiler
 	}
 	target.CFlags = append(append([]string{}, target.osCommon.cflags...), target.CFlags...)
+	target.CFlags = append(append([]string{}, target.cflagsCommon...), target.CFlags...)
```

Comment:
It becomes quite hard to figure the resulting order of flags when reading this and the syntax become quite bloated.
I propose something along the following lines:
```
for _, flags := range [][]string{commonCFlags, target.cflagsCommon, target.osCommon.cflags, target.cflags} {
	target.CFlags = append(target.CFlags, flags)
}
```

Reply:
Given that we're only appending three arrays, do you still want this change?
I.e. 
```
-       target.CFlags = append(append([]string{}, target.osCommon.cflags...), target.CFlags...)
-       target.CFlags = append(append([]string{}, commonCFlags...), target.CFlags...)
+       for _, flags := range [][]string{commonCFlags, target.osCommon.cflags} {
+               target.CFlags = append(target.CFlags, flags)
+       }
```